### ISSUE TITLE
CodeReady Containers (CRC) disconnected install -- howto?

### ISSUE THREAD

**Issue by pthopa** [code-ready/crc#1004]
### General information

  * OS: **RHEL 7.6**
  * Hypervisor: **KVM** 
  * Did you run `crc setup` before starting it (Yes/No)? 
   **Yes**

## CRC version
```bash
# Put the output of `crc version`
```
**crc version: 1.5.0+e1c8fb8
Openshift version: 4.2.14**
  
## CRC status
```bash
# Put the output of `crc status`
```
CRC VM:Running
Openshift: Runnin g=g (v4.2.14-x86_64)
Disk Usage: 17.71GB of 32.2GB
Cache Usage: 11.98GB
Cache Directory: ~/.crc/cache

## CRC config
```bash
# Put the output of `crc config view`
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
```

### Steps to reproduce

Unable to deploy and run applications  in disconnected network. Not able to find a clear set of instructions/documents on how to do this.

### Expected
Clear documentation/steps on how to run crc in a disconnected network.

### Logs

You can start crc with `crc start --log-level debug` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by cfergeau** [comment]
https://docs.openshift.com/container-platform/4.3/installing/install_config/installing-restricted-networks-preparations.html has some instructions regarding a disconnected openshift install, but in crc's case that's indeed a different setup

**Comment by pthopa** [comment]
@cfergeau  thanks.. yes i did take a look at that..  and also @ https://github.com/code-ready/snc/issues/154...  in that praveen kumar mentions that the "deployer" image has to be added manually to crc..
i have asked him how to do that

**Comment by pthopa** [comment]
i found this link for minishift disconnected install...  http://blog.andyserver.com/2018/10/minishift-and-the-enterprise-disconnected-image-registry/

**Comment by gbraad** [comment]
CDK is based on OpenShift 3.x and therefore not usable to get this working. Minishift/CDK came even with an addon that sets up the environment.

**Comment by praveenkumar** [comment]
@pthopa what you are asking is restricted environment not the disconnected (when there is not internet connection). As per the docs which shared by @cfergeau is mostly from installation time, we are still try to figure out if there is any document exist which point out if you cluster is already installed then how you can move this to restricted environment.

**Comment by pthopa** [comment]
1. My colleague followed instructions @ https://docs.openshift.com/container-platform/4.2/installing/install_config/installing-restricted-networks-preparations.html
to create a mirror registry.
2. i then created an ImageContentSourcePolicy to mirror requests for "quay.io/openshift-release-dev/ocp-v4.0-art-dev" to our mirrored registry..
3. But still openshift seems to ignore the mirror configuration and still tries to pull from quay.io.

I saw a bug  https://bugzilla.redhat.com/show_bug.cgi?id=1745192
(Bug 1745192 - Builds are not configured to use mirrors in disconnected enivronments)

So still stuck with getting crc to work.

**Comment by praveenkumar** [comment]
@pthopa we released the 1.6.0 which is based on 4.3.0 can you try using that and let us know if that solve your issue?

**Comment by pthopa** [comment]
yes.. we are using the latest version of crc and our issue is resolved now
